### PR TITLE
Add Line TKL keyboard.

### PR DESCRIPTION
**Pull request by studiokestra**
## Description

Add support for LINE FRIENDS TKL PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/studiokestra/line_tkl/info.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        "pid": "0x8704",
+        "vid": "0x7C10"
+    },
+    "indicators": {
+        "caps_lock": "B6",
+        "scroll_lock": "D2",
+        "on_state": 1
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/info.json:32]: could a layout with 2u Backspace and 2.75u Right shift (`LAYOUT_tkl_ansi_tsangan`) be added?

File: keyboards/studiokestra/line_tkl/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    [0] = LAYOUT(
+    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                         KC_PSCR, KC_SCRL, KC_PAUS,
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_BSPC,     KC_INS,  KC_HOME, KC_PGUP,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,               KC_DEL,  KC_END,  KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,     
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                                   KC_UP,          
+    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,                        KC_LEFT, KC_DOWN, KC_RIGHT ),
+
+    [1] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/keymaps/default/keymap.c:32]:
```suggestion
    [0] = LAYOUT_tkl_ansi_tsangan_split_bs_rshift(
    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                         KC_PSCR, KC_SCRL, KC_PAUS,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_BSPC,     KC_INS,  KC_HOME, KC_PGUP,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,               KC_DEL,  KC_END,  KC_PGDN,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,     
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                                   KC_UP,          
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,                        KC_LEFT, KC_DOWN, KC_RIGHT ),

    [1] = LAYOUT_tkl_ansi_tsangan_split_bs_rshift(
```

File: keyboards/studiokestra/line_tkl/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    [0] = LAYOUT(
+    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                         KC_PSCR, KC_SCRL, KC_PAUS,
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_BSPC,     KC_INS,  KC_HOME, KC_PGUP,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,               KC_DEL,  KC_END,  KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,     
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                                   KC_UP,          
+    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,                        KC_LEFT, KC_DOWN, KC_RIGHT ),
+
+    [1] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/keymaps/via/keymap.c:32]:
```suggestion
    [0] = LAYOUT_tkl_ansi_tsangan_split_bs_rshift(
    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                         KC_PSCR, KC_SCRL, KC_PAUS,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_BSPC,     KC_INS,  KC_HOME, KC_PGUP,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,               KC_DEL,  KC_END,  KC_PGDN,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,     
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                                   KC_UP,          
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL,                        KC_LEFT, KC_DOWN, KC_RIGHT ),

    [1] = LAYOUT_tkl_ansi_tsangan_split_bs_rshift(
```

File: keyboards/studiokestra/line_tkl/info.json (hunk 1 above)
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/info.json:32]:
```suggestion
    },
    "community_layouts": ["tkl_ansi_tsangan_split_bs_rshift"],
    "layouts": {
        "LAYOUT_tkl_ansi_tsangan_split_bs_rshift": {
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by studiokestra** [comment]
Added tkl_ansi_tsangan layout and set to default. added split_bs_rshift layout

File: keyboards/studiokestra/line_tkl/info.json [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    "indicators": {
+        "caps_lock": "B6",
+        "scroll_lock": "D2",
+        "on_state": 1
+    },
+    "community_layouts": ["tkl_ansi_tsangan","tkl_ansi_tsangan_split_bs_rshift"],
+    "layouts": {
+        "LAYOUT_tkl_ansi_tsangan": {
+            "layout": [
+
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/info.json:35]: tidy
```suggestion
    "community_layouts": [
        "tkl_ansi_tsangan",
        "tkl_ansi_tsangan_split_bs_rshift"
    ],
    "layouts": {
        "LAYOUT_tkl_ansi_tsangan": {
            "layout": [
```

File: keyboards/studiokestra/line_tkl/info.json [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+                { "matrix": [10,6], "x":12.5,"y":5.25}, 
+                { "matrix": [10,7], "x":13.5,"y":5.25,"w":1.5},
+                { "matrix": [11,7], "x":15.25, "y":5.25}, 
+                { "matrix": [10,8], "x":16.25, "y":5.25}, 
+                { "matrix": [11,8], "x":17.25, "y":5.25}
+            ]
+        },
+        "LAYOUT_tkl_ansi_tsangan_split_bs_rshift": {
+            "layout": [
+
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/info.json:126]: tidy
```suggestion
```

File: keyboards/studiokestra/line_tkl/info.json [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+                { "matrix": [11,5], "x":11,"y":5.25,"w":1.5}, 
+                { "matrix": [10,6], "x":12.5,"y":5.25}, 
+                { "matrix": [10,7], "x":13.5,"y":5.25,"w":1.5},
+                { "matrix": [11,7], "x":15.25, "y":5.25}, 
+                { "matrix": [10,8], "x":16.25, "y":5.25}, 
+                { "matrix": [11,8], "x":17.25, "y":5.25}
+            ]
+        }
+    }
+}
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/info.json:218]: add EoF
```suggestion
}

```

File: keyboards/studiokestra/line_tkl/config.h [hunk 7]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2023 studiokestra (@studiokestra)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (dunk2k) [inline keyboards/studiokestra/line_tkl/config.h:9]: add EoF
```suggestion
#define LOCKING_RESYNC_ENABLE

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**